internal/compiler/pathutils.rs

```diff
@@ -0,0 +1,528 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use std::path::{Path, PathBuf};
+
+// Reimplement some Path handling code: The one in `std` is not available
+// when running in WASM!
+//
+// This is not helped by us using URLs in place of paths *sometimes*.
+
+pub fn is_url(path: &Path) -> bool {
+    let Some(path) = path.to_str() else {
+        // URLs can always convert to string in Rust
+        return false;
+    };
+
+    let Ok(url) = url::Url::parse(path) else {
+        return false;
+    };
+    url.scheme().len() > 1
+}
+
+pub fn is_absolute(path: &Path) -> bool {
+    if is_url(path) {
```

Comment:
is_url is fairly expensive as it parses the URL, Can't your parsing logic detect the root as a URL anyway? (It does match ":/". I think the wasm std::Path does it properly.

Reply:
But both are absolute path, so i don't see why this is a problem.